plugins/org.komodo.relational/src/org/komodo/relational/internal/RelationalObjectImpl.java

```diff
@@ -34,6 +35,19 @@ protected RelationalObjectImpl( final Repository repository,
     /**
      * {@inheritDoc}
      *
+     * @see org.komodo.repository.ObjectImpl#setPrimaryType(org.komodo.spi.repository.Repository.UnitOfWork, java.lang.String)
+     * @throws UnsupportedOperationException
+     *         if called
+     */
+    @Override
+    public final void setPrimaryType( final UnitOfWork uow,
+                                      final String typeName ) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
```

Comment:
I like putting the runtime exceptions in the signature just so that someone looking at the API could see that it does throw an exception. This is just personal preference. I added this method here, in `RelationModelObjectImpl`, as more of an "assert" to us so that we know we shouldn't be calling it. But I have no problems changing it to a `KException`.